src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1723,8 +1742,207 @@ private function step_in_body() {
 	 *
 	 * @return bool Whether an element was found.
 	 */
-	private function step_in_table() {
-		$this->bail( "No support for parsing in the '{$this->state->insertion_mode}' state." );
+	private function step_in_table(): bool {
+		$token_name = $this->get_token_name();
+		$token_type = $this->get_token_type();
+		$op_sigil   = '#tag' === $token_type ? ( parent::is_tag_closer() ? '-' : '+' ) : '';
+		$op         = "{$op_sigil}{$token_name}";
+
+		switch ( $op ) {
+			/*
+			 * > A character token, if the current node is table, tbody, template, tfoot, thead, or tr element
+			 */
+			case '#text':
+				if (
+					$this->state->stack_of_open_elements->current_node_is( 'TABLE' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TBODY' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TEMPLATE' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TFOOT' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'THEAD' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TR' )
+				) {
+					$text = str_replace( "\0", '', $this->get_modifiable_text() );
+					// Ignore text nodes that are entirely U+0000 NULL.
+					if ( '' === $text ) {
+						// @todo Indicate a parse error once it's possible.
+						return $this->step();
+					}
+
+					// Whitespace-only text nodes are inserted in-place.
+					if ( strlen( $text ) === strspn( $text, "\u{0009}\u{000A}\u{000C}\u{000D}\u{0020}", 0, strlen( $text ) ) ) {
```

Comment:
```suggestion
					/*
					 * This follows the rules for "in table text" insertion mode.
					 *
					 * Whitespace-only text nodes are inserted in-place. Otherwise
					 * foster parenting is enabled and the nodes would be
					 * inserted out-of-place.
					 *
					 * > If any of the tokens in the pending table character tokens
                                         * > list are character tokens that are not ASCII whitespace,
                                         * > then this is a parse error: reprocess the character tokens
                                         * > in the pending table character tokens list using the rules
                                         * > given in the "anything else" entry in the "in table"
                                         * > insertion mode.
                                         * >
                                         * > Otherwise, insert the characters given by the pending table
                                         * > character tokens list.
					 *
					 * @see https://html.spec.whatwg.org/#parsing-main-intabletext
                                         */
					if ( strlen( $text ) === strspn( $text, "\x09\x0A\x0C\x0D\x20", 0, strlen( $text ) ) ) {
```